R/locations_and_mascots.R

```diff
@@ -0,0 +1,233 @@
+## LOCATIONS AND MASCOTS (by: pep)
+## for: MLB, NBA, NCAA, NFL, and NHL (NOT: EPL or MLS)
+
+## please run the ncaa_color_update script
+## please run the logos.R script
+## make sure to load rvest after tidyverse (purr)
+library(tidyverse)
+library(readr)
+library(rvest) 
+
+
+## Removing epl and mls leagues
+
+teamcolors_without_epl_mls <- teamcolors %>%
+  filter(league == "mlb" | league == "nba" |
+           league == "nfl" | league == "nhl" |
+           league == "ncaa")
+
+## removing the weird X1 from the datasets
+teamcolors_without_epl_mls <- teamcolors_without_epl_mls %>%
+  select(-X1)
+
+
+## this is a function that is called on in the name_location_mascot_function below
+## the only thing that this function does is replaces all of the specificed strings to include an underscore
+## this allows the main function to seperate the name of the team into location and mascot, using pluck 
+
+nasty_gsub_function <- function(df){
+  
+  df <- mutate(df,    ## mlb, nfl, nba, nhl
+               
+               name = gsub("Golden State", "Golden_State", name),
+               name = gsub("Green Bay", "Green_Bay", name),
+               name = gsub("Kansas City", "Kansas_City", name),
+               name = gsub("Los Angeles", "Los_Angeles", name),
+               name = gsub("New England", "New_England", name),
```

Comment:
Here, would `gsub("New ", "New_", name)` cover at least these four cases without any false positives? I'd like to make this list as short as possible.